source/widgets/UnixWindow.ooc

```diff
@@ -43,5 +44,15 @@ UnixWindow: class extends DisplayWindow {
 	refresh: override func {
 		this _openGLWindow refresh()
 	}
+	processEvents: override func {
+		while (XPending(this _xWindow display)) {
+			event: XEventOoc
+			XNextEvent(this _xWindow display, event&)
+			if ((event type == ButtonPress) && this _mousePressHandler)
+				this _mousePressHandler call(IntPoint2D new(event xkey x, event xkey y))
+			if ((event type == ButtonRelease) && this _mouseReleaseHandler)
+				this _mouseReleaseHandler call(IntPoint2D new(event xkey x, event xkey y))
+		}
+	}
```

Comment:
Okay, but the problem here is that you're doing things specific to X11 in the class called `UnixWindow`, and you're also importing `inlude/x11`, which is part of `ooc-ui`.